packages/jss/src/index.d.ts

```diff
@@ -7,35 +7,38 @@ import {Properties as CSSProperties} from 'csstype'
 // TODO: refactor to only include Observable types if plugin is installed.
 import {Observable} from 'indefinite-observable'
 
-// TODO: Type data better, currently typed as any for allowing to override it
-type Func<R> = ((data: any) => R)
+type Func<P, T, R> = T extends undefined ? ((data: P) => R) : ((data: P & {theme: T}) => R)
 
-type NormalCssProperties = CSSProperties<string | number>
+type NormalCssProperties = css.Properties<string | number>
 type NormalCssValues<K> = K extends keyof NormalCssProperties
-  ? NormalCssProperties[K]
+  ? NormalCssProperties[K] | JssValue
```

Comment:
This is a bug i resolved this in few PR ago

Reply:
https://github.com/cssinjs/jss/pull/1457

Reply:
@hosseinmd I appreciate your scrutiny. I mentioned in [another comment](https://github.com/cssinjs/jss/pull/1460#discussion_r590701242) that some previous changes got lost after I moved my `node_modules` changes to the forked repository. It seems this was only a problem for `packages/jss/src/index.d.ts`. I re-observed the changes and tried to make sure that only the changes related to _this PR_ were kept. Could you re-examine the diff again and tell me if everything looks good?

Reply:
Note that I cannot stop `prettier` from doing its formatting work, as in the case of `line 13` in the added changes in that file.

Reply:
Looks good to me, thank